apps/core/src/hooks/useGetRollingAverageApys.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useMemo } from 'react';
+
+import { useGetValidatorsEvents } from './useGetValidatorsEvents';
+import { roundFloat } from '../utils/roundFloat';
+
+// recentEpochRewards is list of the last 30 epoch rewards for a specific validator
+// APY_e = (1 + epoch_rewards / stake)^365-1
+// APY_e_30rollingaverage = average(APY_e,APY_e-1,…,APY_e-29);
+
+const ROLLING_AVERAGE = 30;
+const DEFAULT_APY_DECIMALS = 4;
+
+// define the type parsedJson response
+type ParsedJson = {
+    commission_rate: string;
+    epoch: string;
+    pool_staking_reward: string;
+    pool_token_exchange_rate: {
+        pool_token_amount: string;
+        sui_amount: string;
+    };
+    reference_gas_survey_quote: string;
+    stake: string;
+    storage_fund_staking_reward: string;
+    tallying_rule_global_score: string;
+    tallying_rule_reporters: string[];
+    validator_address: string;
+};
+
+interface ApyGroups {
+    [validatorAddress: string]: number[];
+}
+
+export interface ApyByValidator {
+    [validatorAddress: string]: number;
+}
+
+export function useGetRollingAverageApys(numberOfValidators: number | null) {
+    // Set the limit to the number of validators  * the rolling average
+    // Order the response in descending order so that the most recent epoch are at the top
+    const validatorEpochEvents = useGetValidatorsEvents({
+        limit: numberOfValidators ? numberOfValidators * ROLLING_AVERAGE : null,
```

Comment:
Also are we sure that using limit like this will return always the correct data? Is it possible to include for example events that are from a previous epoch than the `ROLLING_AVERAGE`?

Reply:
The max number of validators we can have is 100,  but realistically max sub 60 * ROLLING_AVERAGE. So ~30 epochs per validator.

Sample data 

`{
                "id": {
                    "txDigest": "47moJQT8FdYLo6qVFTDaTWikKNjkrZkNm97YdqMsZK3v",
                    "eventSeq": 60
                },
                "packageId": "0x0000000000000000000000000000000000000000000000000000000000000002",
                "transactionModule": "sui_system",
                "sender": "0x0000000000000000000000000000000000000000000000000000000000000000",
                "type": "0x2::validator_set::ValidatorEpochInfoEvent",
                "parsedJson": {
                    "commission_rate": "1500",
                    "epoch": "1",
                    "pool_staking_reward": "16666667363783",
                    "pool_token_exchange_rate": {
                        "pool_token_amount": "30002498820106134",
                        "sui_amount": "30016666667363783"
                    },
                    "reference_gas_survey_quote": "1000",
                    "stake": "30016666667363783",
                    "storage_fund_staking_reward": "0",
                    "tallying_rule_global_score": "1",
                    "tallying_rule_reporters": [],
                    "validator_address": "0x8925c11a13cf4b30a64a30ee9f3ca401e58b541b34517d99122e779aa81e3bc9"
                },
                "bcs": "2873gFcH6hDsB3mxQCYuRQQivbMPLWWWLa94pGgtRw9tbzFGQrRMkMYRHwMgToAaFbMjKua1xSH6RKWyMTrrQSJcUeHeUS3JZfkWtDH3XgwTKAZZC9HBA1Yvpqa91FhwX7rGbSSnGG2bE6B",
                "timestampMs": 1678922858372
            }`